### PR TITLE
[BUGFIX canary] Fix _lookupFactory deprecation for Ember canary

### PR DESCRIPTION
**Pull request by locks**

### CODE REVIEW

File: addon/-private/utils.js [hunk 1]
```diff
@@ -41,7 +41,18 @@ function getOwner(context) {
...
 
+/*
+  ember-factory-for is a new feature in Ember 2.12 that finally provides a public API for looking factories up.
+  This function serves as a super simple polyfill to avoid triggering deprecations.
+ */
+function factoryFor(owner) {
+  if (owner.factoryFor) {
+    return owner.factoryFor
+  }
+}
```
Comment (bmac) [inline addon/-private/utils.js]: Is this uses?

File: addon/-private/utils.js [hunk 2]
```diff
@@ -43,5 +43,6 @@ function getOwner(context) {
 
 export {
   modelHasAttributeOrRelationshipNamedType,
-  getOwner
+  getOwner,
+  factoryFor
```
Comment (fivetanley) [inline addon/-private/utils.js]: this appears to not be a function anymore?

Comment (locks) [inline addon/-private/utils.js]: oops, I only removed part of the 💩

File: addon/-private/utils.js [hunk 3]
```diff
@@ -12,11 +12,11 @@ function modelHasAttributeOrRelationshipNamedType(modelClass) {
   return get(modelClass, 'attributes').has('type') || get(modelClass, 'relationshipsByName').has('type');
 }
 
-/*
+/**
```
Comment (bmac) [inline addon/-private/utils.js]: Do you mind removing the extra `*`s that were added in this file? YUIDoc sees these as API docs even though these functions are never made available to the user.

File: addon/-private/system/store.js [hunk 4]
```diff
@@ -2034,7 +2034,14 @@ Store = Service.extend({
     // container = < 1.11
     let owner = getOwner(this);
 
-    let mixin = owner._lookupFactory('mixin:' + normalizedModelName);
+    let mixin;
+
+    if (owner.factoryFor) {
+      mixin = owner.factoryFor(`mixin:${normalizedModelName}`) && owner.factoryFor(`mixin:${normalizedModelName}`).class;
```
Comment (pangratz) [inline addon/-private/system/store.js]: Should this be cached?

Comment (rwjblue) [inline addon/-private/system/store.js]: Seems odd to go through the `factoryFor` twice like this, I'd suggest a refactor to:

```js
let MaybeMixin = owner.factoryFor(`mixin:${normalizedModelName}`);
mixin = MaybeMixin && MaybeMixin.class;
```

File: addon/-private/system/store.js [hunk 5]
```diff
@@ -2107,7 +2114,11 @@ Store = Service.extend({
     let trueModelName = this._classKeyFor(modelName);
     let owner = getOwner(this);
 
-    return owner._lookupFactory(`model:${trueModelName}`);
+    if (owner.factoryFor) {
+      return owner.factoryFor(`model:${trueModelName}`) && owner.factoryFor(`model:${trueModelName}`).class;
```
Comment (pangratz) [inline addon/-private/system/store.js]: This could be cached too

Comment (rwjblue) [inline addon/-private/system/store.js]: Same comment as above, we shouldn't call `factorFor` twice in a row like this...

File: addon/-private/system/store.js [hunk 6]
```diff
@@ -2034,7 +2034,13 @@ Store = Service.extend({
     // container = < 1.11
     let owner = getOwner(this);
 
-    let mixin = owner._lookupFactory('mixin:' + normalizedModelName);
+    if (owner.factoryFor) {
```
Comment (bmac) [inline addon/-private/system/store.js]: Looks like a `let mixin;` got lost in the squash.

Comment (locks) [inline addon/-private/system/store.js]: ✅

**Comment by bmac** [comment]
Thanks @locks.

**Comment by workmanw** [comment]
I was in the process of fixing up ember-data-model-fragments for `ember-data-2.12.0-beta`. I noticed that the beta channel for ember-data is missing this commit. I'm definitely in no hurry, I just wanted to call out that it's labeled `[BUGFIX canary]` but needs to be in the beta channel.

**Comment by rwjblue** [comment]
Ahh, good catch @workmanw. It was likely still canary when the PR was proposed, but once we branched beta for 2.12 the commit prefix didn't get updated. 

@bmac - Can this be pulled in for 2.12 beta cycle?

**Comment by bmac** [comment]
Will do.

**Comment by bmac** [comment]
This pr has been pulled into the beta cycle and included in the https://github.com/emberjs/data/releases/tag/v2.12.0-beta.2 release.

**Comment by rwjblue** [comment]
Thank you @bmac!

**Comment by workmanw** [comment]
I'm happy to report ember-data-model-fragments' test pass on all channels. Thanks!!

**Pull request merged**